antenna/metrics.py

```diff
@@ -183,13 +200,28 @@ def __init__(self, name):
         # Remove . at beginning and end
         self.name = name.strip('.')
 
+        self.batch = {}
+
     def _full_stat(self, stat):
         return self.name + '.' + stat
 
     def incr(self, stat, value=1):
         """Increment a stat by value"""
         _metrics_impl.incr(self._full_stat(stat), value=value)
 
+    def batch_incr(self, stat, value=1):
+        """Batches a count for a stat by value
+
+        This is the same as ``.incr()``, however it gets batched and sent out
+        every heartbeat_interval.
+
+        This is helpful for counters that count more frequently than 1/s where
+        statsd might "do interesting things" such that you don't get good
+        counts.
+
+        """
+        _metrics_impl.incr(self._full_stat(stat), value=value)
```

Comment:
We could do the batching/flushing in this class, but ... there's possibly one of these for every component in Antenna. Instead, I decided to do it in the implementation since there's only one of those at any given time.